#51 obtain the leaf index of gbdt tree

grasevski opened the pull request.

dmitryikh commented on lgensemble.go:
```diff
@@ -52,6 +52,13 @@ func (e *lgEnsemble) predictInner(fvals []float64, nEstimators int, predictions
 	}
 }
 
+func (e *lgEnsemble) predictLeaves(fvals []float64, predictions []int) error {
```
seems like we don't need to return `error` here

dmitryikh commented on xgensemble.go:
```diff
@@ -57,6 +57,13 @@ func (e *xgEnsemble) predictInner(fvals []float64, nEstimators int, predictions
 	}
 }
 
+func (e *xgEnsemble) predictLeaves(fvals []float64, predictions []int) error {
```
we don't need to return error here?

grasevski commented:
OK I've addressed your comments and only implemented for lightgbm as it seems the leaves aren't reordered in that implementation from what I can see. Also I'm only using lightgbm so that is ok for my use case.

dmitryikh commented:
done in #76.
